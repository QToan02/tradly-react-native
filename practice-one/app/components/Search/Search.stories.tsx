import { useState } from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react-native'

import { Search } from '../index'
import { SearchProps } from './index'

export default {
  title: 'Search',
  component: Search,
} as ComponentMeta<typeof Search>

const Template: ComponentStory<typeof Search> = (
  args: Omit<SearchProps, 'value' | 'onChangeText'>
) => {
  const [value, onChangeValue] = useState<string>('')

  return <Search value={value} onChangeText={onChangeValue} {...args} />
}

export const Default = Template.bind({})
Default.args = {
  placeholder: 'Search Product',
}
