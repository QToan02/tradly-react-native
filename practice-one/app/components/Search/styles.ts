import { StyleSheet } from 'react-native'

import { COLORS, FONT_FAMILY } from '@constants'

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    columnGap: 20,
    borderRadius: 50,
    paddingLeft: 17,
    backgroundColor: COLORS.WHITE,
  },
  icon: {
    width: 24,
    height: 24,
    resizeMode: 'cover',
  },
  search: {
    flex: 1,
    fontFamily: FONT_FAMILY.MONTSERRAT[500],
    fontSize: 14,
    color: COLORS.GRAY_50,
    paddingVertical: 8,
  },
})

export default styles
