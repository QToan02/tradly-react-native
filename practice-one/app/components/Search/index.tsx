import { useRef } from 'react'
import { Image, TextInput, TextInputProps, TouchableWithoutFeedback, View } from 'react-native'

import { COLORS } from '@constants'
import styles from './styles'

export interface SearchProps extends TextInputProps {
  value: string
  placeholder: string
  onChangeText: (text: string) => void
}

const Search = ({ value, placeholder, onChangeText, ...rest }: SearchProps) => {
  const searchInput = useRef<TextInput>(null)
  const handlePress = () => {
    searchInput.current?.focus()
  }

  return (
    <TouchableWithoutFeedback onPress={handlePress}>
      <View style={styles.container}>
        <Image style={styles.icon} source={require('@assets/search.png')} />
        <TextInput
          ref={searchInput}
          style={styles.search}
          placeholder={placeholder}
          placeholderTextColor={COLORS.GRAY_50}
          value={value}
          onChangeText={onChangeText}
          {...rest}
        />
      </View>
    </TouchableWithoutFeedback>
  )
}

export default Search
