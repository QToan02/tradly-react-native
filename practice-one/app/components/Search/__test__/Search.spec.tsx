import { useState } from 'react'
import { render, fireEvent, renderHook } from '@testing-library/react-native'

import Search from '../index'

describe('Testing Search component', () => {
  it('Render Search component success', () => {
    const { result } = renderHook(() => useState(''))
    const [value, onChangeValue] = result.current
    const component = render(
      <Search value={value} onChangeText={onChangeValue} placeholder="my search input" />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('When change input value onChangeText must be called with right props', () => {
    const mockOnChange = jest.fn()
    const { getByPlaceholderText } = render(
      <Search value="my first value" onChangeText={mockOnChange} placeholder="my search input" />
    )
    const element = getByPlaceholderText('my search input')

    fireEvent.changeText(element, 'new products')
    expect(mockOnChange).toBeCalledWith('new products')
  })
})
