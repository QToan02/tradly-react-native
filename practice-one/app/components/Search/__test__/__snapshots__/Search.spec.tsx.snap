// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Search component Render Search component success 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 50,
      "columnGap": 20,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingLeft": 17,
      "paddingVertical": 10,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/search.png",
      }
    }
    style={
      {
        "height": 24,
        "resizeMode": "cover",
        "width": 24,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="my search input"
    placeholderTextColor="#4f4f4f"
    style={
      {
        "color": "#000",
        "flex": 1,
        "fontFamily": "Montserrat_500Medium",
        "fontSize": 14,
      }
    }
    value=""
  />
</View>
`;
